src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.iOS.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System.Globalization
+{
+    internal sealed partial class CultureData
+    {
+        // Native constants, check if we need this for native
+        private const int Native_ULOC_KEYWORD_AND_VALUES_CAPACITY = 100; // max size of keyword or value
+        private const int Native_ULOC_FULLNAME_CAPACITY = 157;           // max size of locale name
+
+        /// <summary>
+        /// This method uses the sRealName field (which is initialized by the constructor before this is called) to
+        /// initialize the rest of the state of CultureData based on the underlying OS globalization library.
+        /// </summary>
+        private bool InitNativeCultureDataCore()
+        {
+            Debug.Assert(_sRealName != null);
+            Debug.Assert(!GlobalizationMode.Invariant);
+            string realNameBuffer = _sRealName;
+
+            // Get the locale name
+            NativeGetLocaleName(realNameBuffer, out _sWindowsName);
+            return true;
+        }
+
+        internal static unsafe bool NativeGetLocaleName(string localeName, out string? windowsName)
+        {
+            windowsName = Interop.Globalization.NativeGetLocaleName(localeName, Native_ULOC_FULLNAME_CAPACITY);
+            return true;
+        }
+
+        private string NativeGetLocaleInfo(LocaleStringData type, string? uiCultureName = null)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
```

Comment:
Out of curiosity, why is it expected that GlobalizationMode.Invariant and GlobalizationMode.UseNls both be false here? Are there general expectations that both `GlobalizationMode.Invariant` and `GlobalizationMode.Hybrid` cannot be both true, or are they completely orthogonal?

Reply:
For UseNls: GlobalizationMode.UseNls is always false on iOS. It is Windows-specific property. I will remove checks for iOS.
For Invariant: It means all globalization specific code and data is trimmed when set to true and icu is not loaded. Hybrid means we are still loading icu + using native functions. Both Invariant and hybrid cannot be true.